rules.js

```diff
@@ -1044,7 +1047,17 @@ function tryActionCore_(player, action /*Action object*/) {
     player.bridges++;
   }
   else if(action.type == A_PLACE_BRIDGE) {
-    if(player.bridges <= 0) return 'not enough bridges';
+    //LOU CHEAT if get here after A_POWER_BRIDGE fails     
+    if (player.bridges <= 0) {
```

Comment:
So what this does is restore the player resources?

That should actually not really be here I think - it's just trying to fix a symptom of another bug.

Either a bug in the AI, for trying to place a bridge while it's impossible (it shouldn't, so that would be a fix in strategy.js and/or ai.js)
Or a bug in the game engine for not properly restoring resources after a failed action. I will look into that case later.

Thanks :)